src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/PropertyAccessor.cs

```diff
@@ -117,9 +130,79 @@ public abstract class Accessor
             public bool IsGroupByField;
         }
 
-        internal static PropertyAccessor CreateMapReduceOutputAccessor(Type type, HashSet<string> _groupByFields)
+        internal static IPropertyAccessor CreateMapReduceOutputAccessor(Type type, HashSet<string> _groupByFields, bool isObjectInstance = false)
         {
+            if (isObjectInstance || type == typeof(ObjectInstance))
+                return new JintPropertyAccessor(_groupByFields);
+
             return new PropertyAccessor(type, _groupByFields);
         }
+
+    }
+
+    internal class JintPropertyAccessor : IPropertyAccessor
+    {
+        private HashSet<string> _groupByFields;
+
+        public JintPropertyAccessor(HashSet<string> groupByFields)
+        {
+            _groupByFields = groupByFields;
+        }
+
+        public IEnumerable<(string Key, object Value, bool IsGroupByField)> GetPropertiesInOrder(object target)
+        {
+            var oi = target as ObjectInstance;
+            if(oi == null)
+                throw new ArgumentException($"JintPropertyAccessor.GetPropertiesInOrder is expecting a target of type ObjectInstance but got one of type {target.GetType().Name}.");
+            foreach (var property in oi.GetOwnProperties())
+            {
+                yield return (property.Key, GetValue(property.Value.Value), _groupByFields?.Contains(property.Key)??false);
+            }            
+        }
+
+        public object GetValue(string name, object target)
+        {
+            var oi = target as ObjectInstance;
+            if (oi == null)
+                throw new ArgumentException($"JintPropertyAccessor.GetValue is expecting a target of type ObjectInstance but got one of type {target.GetType().Name}.");
+            if(oi.HasOwnProperty(name) == false)
+                throw new MissingFieldException($"The target for 'JintPropertyAccessor.GetValue' doesn't contain the property {name}.");
+            return GetValue(oi.GetProperty(name).Value);
+        }
+
+        private static object GetValue(JsValue jsValue)
+        {
+            if (jsValue.IsNull())
+                return null;
+            if (jsValue.IsString())
+                return jsValue.AsString();
+            if (jsValue.IsBoolean())
+                return jsValue.AsBoolean(); 
+            if (jsValue.IsNumber())
+                return jsValue.AsNumber();
+            if (jsValue.IsDate())
+                return jsValue.AsDate();
+            if (jsValue.IsObject())
+            {
+                return jsValue.AsObject();
```

Comment:
How useful is this? It still doesn't know how to handle this since this is not primitive